SW_VegProd.c

```diff
@@ -841,3 +865,541 @@ void get_critical_rank(void){
+                            overallEstimSize--;
+                        }
+                    } while(index != overallEstimSize - 1 && isGrassIndex);
+                }
+                overallEstimSize--;
+            }
+
+            // Get final estimated vegetation sum
+            for(index = 0; index < nTypes; index++) {
+                if(missing(inputValues[index])) {
+                    finalVegSum += estimCover[index];
+                } else {
+                    finalVegSum += inputValues[index];
+                    fixedValuesSum += inputValues[index];
+                }
+            }
+
+            // Check if the final estimated vegetation sum is equal to one
+            if(!EQ(finalVegSum, 1.)) {
+                for(index = 0; index < overallEstimSize; index++) {
+                    estimCoverSum += estimCover[overallEstim[index]];
+                }
+                if(estimCoverSum > 0) {
+                    for(index = 0; index < overallEstimSize; index++) {
+                        estimCover[overallEstim[index]] *= (1 - fixedValuesSum) / estimCoverSum;
+                    }
+                } else {
+                    if(fillEmptyWithBareGround && EQ(inputValues[bareGround], 0.0)) {
+                        estimCover[bareGround] = 1.;
+                        for(index = 0; index < nTypes - 1; index++) {
+                            estimCover[bareGround] -= estimCover[index];
+                        }
+                    } else {
+                        LogError(logfp, LOGFATAL, "'estimate_PotNatVeg_composition': "
+                                    "The estimated vegetation cover values are 0, "
+                                    "the user fixed relative abundance values sum to less than 1, "
+                                    "and bare-ground is fixed. "
+                                    "Thus, the function cannot compute "
+                                    "complete land cover composition.");
+                    }
+                }
+            }
+        }
+    }
+
+    // Fill in all output arrays (grassOutput, RelAbundanceL0, RelAbundanceL1)
+    for(index = 0; index < nTypes; index++) {
+        if(!missing(inputValues[index]) && index != bareGround) {
+            estimCover[index] = inputValues[index];
+        }
+
+        RelAbundanceL0[index] = estimCover[index];
+    }
+
+    grassOutput[0] = estimCover[C3Index];
+    grassOutput[1] = estimCover[C4Index];
+    grassOutput[2] = estimCover[grassAnn];
+
+    tempSumGrasses = mean(grassOutput, 3) * 3;
+    if(tempSumGrasses > 0) {
+        for(index = 0; index < 3; index++) {
+            grassOutput[index] /= tempSumGrasses;
+        }
+    }
+
+    RelAbundanceL1[0] = estimCover[treeIndex];
+    RelAbundanceL1[1] = estimCover[shrubIndex];
+    RelAbundanceL1[2] = estimCover[forbIndex] + estimCover[succIndex];
+    RelAbundanceL1[3] = tempSumGrasses;
+    RelAbundanceL1[4] = estimCover[bareGround];

```

Comment:
If `inputValues[bareGround] = SW_MISSING` then incorrectly `RelAbundanceL1[4] = SW_MISSING` instead of `RelAbundanceL1[4] = RelAbundanceL0[bareGround]`

Reply:
`inputValues[bareGround] = SW_MISSING` cannot happen, "inputValues" fixes bare ground value to 0.0 by default.

Reply:
fixed by 4a7b06dfdbe9229c5cce15bd62eb15e1e2caf640